xmtp_mls/src/storage/encrypted_store/group_message.rs

```diff
@@ -130,6 +167,41 @@ impl DbConnection<'_> {
                 .optional()
         })?)
     }
+
+    pub fn set_delivery_status_to_published<MessageId: AsRef<[u8]>>(
+        &self,
+        msg_id: &MessageId,
+    ) -> Result<usize, StorageError> {
+        let res = self.raw_query(|conn| {
+            diesel::update(dsl::group_messages)
+                .filter(dsl::id.eq(msg_id.as_ref()))
+                .set(dsl::delivery_status.eq(DeliveryStatus::Published))
+                .execute(conn)
+        });
+
+        match res {
+            Ok(n) => Ok(n),
+            Err(_e) => Err(StorageError::NotFound),
```

Comment:
You can verify via unit test, but I don't think this method and the one below will error if nothing was found. Under the hood the sql is just doing `update WHERE`. There are some other examples in the codebase, but the `Ok` variant of `res` should return the number of updates, you just need to verify that it's equal to 1.

Reply:
you are absolutely right.  I looked at for 10 seconds before realizing we don't need the `match res {...}` block at all.  Replaced with a `?` above, which should bubble up the correct variant of `StorageError`, if any.